src/components/featured-products-section/featured-products-section.tsx

```diff
@@ -0,0 +1,83 @@
+import { useEffect, useState } from 'react';
+import { ProductCard } from '../product-card/product-card';
+import { ProductLink } from '../product-link/product-link';
+import styles from './featured-products-section.module.scss';
+import { collections } from '@wix/stores';
+import { getEcomApi } from '~/api/ecom-api';
+import { Product } from '@wix/stores_products';
+import classNames from 'classnames';
+
+interface FeaturedProductsData {
+    category: collections.Collection;
+    products: Product[];
+}
+
+const getFeaturedProducts = async (
+    categorySlug: string,
+    limit: number
+): Promise<FeaturedProductsData | null> => {
+    const api = getEcomApi();
+
+    let category: collections.Collection;
+    try {
+        category = await api.getCategoryBySlug(categorySlug);
+    } catch (error) {
+        try {
+            category = await api.getCategoryBySlug('all-products');
```

Comment:
I think this should check the error type more specifically. We want to fetch all products if the category doesn't exist. In case of any other error we want to throw it.

Reply:
@ynesteruk 's PR #34 has new code in this function:
```
if (isEcomSDKError(e) && e.details.applicationError.code === 404) {
    return failureResponse(EcomApiErrorCodes.CategoryNotFound);
}
```
I will need to update my function. Or @ynesteruk. Depends who merge first :)